lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -74,11 +74,11 @@ namespace {
   /// condition is empty.
   struct SpaceEngine {
     enum class SpaceKind : uint8_t {
-      Empty           = 1 << 0,
```

Comment:
These aren't just tags, they're powering the `constexpr` bit-mangling/pattern-matching gadget in the rest of this file.  With these values you get easy uniqueness of bit sequences closed under `|`.

Reply:
It works either way, I guess.

Reply:
That doesn't make sense to me. Your pairwise case works for *any* two uint8_ts, and that's already encoded in the enum's underlying type. And indeed, it would *not* work for arbitrary flags, and so making the constants powers of two is misleading.

Reply:
> That doesn't make sense to me.

Let me be clearer: By "easy" I mean I could dump the binary value of a pair of spaces and easily see from the bits which ones I had.  Either way works for me, but if it's misleading to treat it as an option set I would still like the tag values to be explicitly stated in the source.

Reply:
We don't do that for any other enums except where the raw value is significant, so again it seems weird to include the values (especially when there's only 5 of them and the debugger can print them for you). I'm  also not sure how this is really any different with regards to dumping. Before you'd get values like `0x0408`, `0x0210`, etc; now you get `0x0203` and `0x0104`.